File: Data Structures & Algorithms/longest-common-prefix/submission-0.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        for i, c in enumerate(strs[0]):
            for j in range(1, len(strs)):
                if i > len(strs[j]) - 1 or c != strs[j][i]: 
                    return strs[0][0:i]
        return strs[0]

File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> anagramGroups = new HashMap<>();
        for (String s: strs) {
            String sortedStr = Stream.of(s.split(""))
            .sorted()
            .collect(Collectors.joining());
            if (anagramGroups.containsKey(sortedStr)) {
                anagramGroups.get(sortedStr).add(s);
            }
            else {
                List<String> values = new  ArrayList<>();
                values.add(s);
                anagramGroups.put(sortedStr, values);
            }
        }
        return anagramGroups.values().stream().collect(Collectors.toList());
    }
}
